istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,714 @@
+	}
+	if opts.disableValidationWebhook && opts.validatingWebhookConfigName == "" {
+		return fmt.Errorf("validating webhook config to disable has empty name")
+	}
+	if opts.disableInjectionWebhook && opts.mutatingWebhookConfigName == "" {
+		return fmt.Errorf("mutating webhook config to disable has empty name")
+	}
+	return nil
+}
+
+type statusCliOptions struct {
+	// Whether display the webhook configuration of Galley
+	validationWebhook bool
+	// Whether display the webhook configuration of Sidecar Injector
+	injectionWebhook bool
+	// The name of the ValidatingWebhookConfiguration to manage
+	validatingWebhookConfigName string
+	// The name of the MutatingWebhookConfiguration to manage
+	mutatingWebhookConfigName string
+}
+
+// Validate return nil if no error. Otherwise, return the error.
+func (opts *statusCliOptions) Validate() error {
+	if !opts.validationWebhook && !opts.injectionWebhook {
+		return fmt.Errorf("no webhook configuration to display")
+	}
+	if opts.validationWebhook && opts.validatingWebhookConfigName == "" {
+		return fmt.Errorf("validating webhook config to display has empty name")
+	}
+	if opts.injectionWebhook && opts.mutatingWebhookConfigName == "" {
+		return fmt.Errorf("mutating webhook config to display has empty name")
+	}
+	return nil
+}
+
+// Webhook command to manage webhook configurations
+func Webhook() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "webhook",
+		Short: "webhook command to manage webhook configurations",
+	}
+
+	cmd.AddCommand(newEnableCmd())
+	cmd.AddCommand(newDisableCmd())
+	cmd.AddCommand(newStatusCmd())
+
+	return cmd
+}
+
+func newEnableCmd() *cobra.Command {
+	opts := &enableCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable webhook configurations",
+		Example: `
+# Enable the webhook configuration of Galley with the given webhook configuration
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --validation-path validatingwebhookconfiguration.yaml
+
+# Enable the webhook configuration of Galley with the given webhook configuration and CA certificate
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --validation-path validatingwebhookconfiguration.yaml --ca-bundle-file ./k8s-ca-cert.pem
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// --namespace is used as the namespace of the webhook secret
+			opts.webhookSecretNameSpace = namespace
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating Kubernetes client interface: %v", err)
+			}
+			err = enableWebhookConfig(client, opts)
+			if err != nil {
+				return fmt.Errorf("err when enabling webhook configurations: %v", err)
+			}
+			fmt.Println("webhook configurations have been enabled")
```

Comment:
Don't use fmt.Println() nor fmt.Printf().  Use Cobra's cmd.Print functions.  These are easier to test and redirect.

Reply:
Done.